src/dos/dos_locale.cpp

```diff
@@ -0,0 +1,2904 @@
+/*
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ *  Copyright (C) 2020-2023  The DOSBox Staging Team
+ *  Copyright (C) 2002-2021  The DOSBox Team
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+#include "dos_locale.h"
+
+#include <cstring>
+#include <map>
+#include <vector>
+
+#if defined(WIN32)
+#include <windows.h>
+#endif
+
+#include "bitops.h"
+#include "checks.h"
+#include "dos_inc.h"
+#include "dos_keyboard_layout.h"
+#include "logging.h"
+#include "string_utils.h"
+
+CHECK_NARROWING();
+
+// NOTE: Locale settings below were selected based on our knowledge and various
+// public sources. If in doubt, we choose what we think is relevant for the most
+// representative languange for the given country.
+// We do not mean to stigmatise any minorities, we do not mean to offend anyone,
+// we do our best to put our personal opinions and beliefs aside.
```

Comment:
A bit too touchy-feely for my taste, but if this preempts some drama, I guess some tiptoeing doesn't matter. You'd think no one would be upset about a slightly wrong localisation setting in an obscure DOS emulator, but nations went to war for less... 🥶 🤦🏻